src/common/search/FilterPlays.jsx

```diff
@@ -234,19 +244,36 @@ const FilterPlays = ({ reset }) => {
     <div className='search-filter'>
       <Modal
         title='Filter Plays By'
-        onClose={() => setShowModal(false)}
+        onClose={() => {
+          setShowModal(false);
+          const {level_id, tags, owner_user_id, language} = modifiedFilterQuery;
+          const isFilterEmpty= level_id !== "" || tags.length !== 0 || owner_user_id !== "" || language !== "";
+          // if user closes modal instead of clicking on Apply after clear All filters
+          if(!isFilterApplied && !isFilterEmpty) {
+            setModifiedFilterQuery({...filterQuery});
+            setnoOfAppliedFilter(getAppliedFilter(filterQuery));
+          }
+        }}
```

Comment:
Can we move this logic to a separate function and use that function here?

Reply:
Sure!
will do that.